test/api/conftest.py

```diff
@@ -0,0 +1,104 @@
+# Copyright 2021-present ScyllaDB
+#
+# This file is part of Scylla.
+#
+# Scylla is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Scylla is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with Scylla.  If not, see <http://www.gnu.org/licenses/>.
+
+# This file configures pytest for all tests in this directory, and also
+# defines common test fixtures for all of them to use. A "fixture" is some
+# setup which an invididual test requires to run; The fixture has setup code
+# and teardown code, and if multiple tests require the same fixture, it can
+# be set up only once - while still allowing the user to run individual tests
+# and automatically setting up the fixtures they need.
+
+import pytest
+import ssl
+
+from requests import Session, Request, Response
+from cassandra.auth import PlainTextAuthProvider
+from cassandra.cluster import Cluster, ConsistencyLevel, ExecutionProfile, EXEC_PROFILE_DEFAULT
+from cassandra.policies import RoundRobinPolicy
+
+# By default, tests run against the CQL server listening
```

Comment:
I found myself unsure whether all the CQL-related stuff - the configuration options, the fixture, etc., is here by mistake (you just copied it) or by design - maybe you plan to add a test in the future which combines CQL requests and API requests?

By the way, if your intention is that **most** tests will combine CQL and API requests, this raises another sacrilegious question: Maybe there shouldn't be a separate REST API test suite at all? Can't we put the REST API tests in the existing cql-pytest? Note that cql-pytest already uses the REST API for some things, such as to implement "nodetool flush",  and other nodetool commands needed for some tests. Nothing really prevents us from adding pure-REST-API tests in cql-pytest.

Although maybe there's also benefit of a separate test suite, to make it more obvious that cql-pytest tests are supposed - mostly - to also work on Cassandra - while none of the REST API tests should. I don't know.

Reply:
Ok, I think both options are reasonable. The separate subdirectory isn't that bad now that the run script is tiny, so maybe it does make sense to separate it. The more I think about it, the more sense it makes to me.

But if you do decide to move it into cql-pytest, you can either put all of them in the same file (e.g., test_rest_api.py), or, you can also create a subdirectory - that works well too - you can see an example in test/cql-pytest/cassandra_tests and its tree of subdirectories.
In that case, check out existing code in test/cql-pytest/nodetool.py that deals with finding the REST API.  Also, regarding "scylla_only" - your new tests can use a fixture or function that returns the rest_api, and if called when the REST API is not available (see example checking code in cql-pytest/nodetool.py) it will call pytest.skip() to skip the test. So you won't need to add a "scylla_only" tag explicitly to each test.

Reply:
okay, it's less work to keep it separated rather than fighting with the cql-pytest ramifications, so I'll leave it that way :)